app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+                )
+            )
+        )
+    }
+
+    private fun buildVoterClassifications(): List<VoterClassification> {
+        val is18OnElectionDay = VoterClassification(
+            type = "eighteen_on_election_day",
+            assertion = newRegistrationData.is18OrOlderByNextElection
+        )
+
+        val isUsCitizen = VoterClassification(
+            type = "united_states_citizen",
+            assertion = newRegistrationData.isUsCitizen
+        )
+
+        val sendCopyInEmail = VoterClassification(
+            type = "send_copy_in_mail", // TODO Is this "I give permission to OSET Org to send me news and updates?" It's looking like it may not be, that might be "opt_in_email"? Where does this come from?
+            assertion = additionalInfoData.hasOptedIntoNewsUpdates
+        )
+
+        val agreedToDeclaration = VoterClassification(
+            type = "agreed_to_declaration", // TODO I can't find where this comes from, but it's in the API docs
+            assertion = true // TODO Defaulting to this until source is confirmed
+        )
+
+        // TODO, this isn't in the docs, but is in the old VoterClassification, should it be?
+        val politicalPartyChanged = VoterClassification(
+            type = "political_party_change",
+            assertion = additionalInfoData.hasChangedPoliticalParty
+        )
+
+        return listOfNotNull(
+            is18OnElectionDay,
+            isUsCitizen,
+            sendCopyInEmail,
+            agreedToDeclaration,
+            politicalPartyChanged
+        )
+    }
+
+    private fun buildVoterIds(): List<VoterId> {
+        val driversLicense = VoterId(
+            type = "drivers_license",
+            // TODO JSON example has this split with spaces, but app doesn't format it, does that matter? eg 99999999 is 99 999 999 in JSON example
+            stringValue = additionalInfoData.pennDotNumber,
+            attestNoSuchId = !additionalInfoData.knowsPennDotNumber
+        )
+
+        val ssn4 = VoterId(
+            type = "ssn4",
+            stringValue = additionalInfoData.ssnLastFour,
+            attestNoSuchId = !additionalInfoData.knowsSsnLastFour
+        )
+
+        // TODO There's a enum value for state_id, but none in the app that I can find. What is the source? Should it be here? It isn't in the old VoterId type
+
+        return listOf(driversLicense, ssn4)
+    }
+
+    private fun buildContactMethods(): List<ContactMethod> {
+        // TODO Do we specify 'fax' anywhere? It's in the API example, but I can't see where we'd derive it from
+        val phoneCapabilities = listOfNotNull(
+            "voice",
+            if (additionalInfoData.phoneType == PhoneType.MOBILE) "sms" else null
+        )
+
+        val phone = ContactMethod(
+            type = "phone",
+            value = additionalInfoData.phoneNumber,
```

Comment:
@huttneab 
I just realized that the JSON example phone number formats are `555-555-5555` whereas the code format is `(555) 555-5555`.

Just calling this out to make sure I'm reading the old code correctly. If I am, no action necessary.

Reply:
yes, the code format is the correct way